blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/astyanax/AstyanaxWriter.java

```diff
@@ -281,6 +327,10 @@ public void insertMetrics(Collection<IMetric> metrics, ColumnFamily cf) throws C
                             Instrumentation.markFullResPreaggregatedMetricWritten();
                         }
                     }
+
+                    if (locatorInsertOk) {
+                        insertDelayedLocator(metric, batch, isTrackingDelayedMetrics, clock);
```

Comment:
We do use cache for inserting delayed locator also similar to the one we have to insert locators. That code is inside that method.